File: src/main/kotlin/base/BaseTest.java
package base;

import driver.PlaywrightManager;
import utils.ScreenshotUtil;
import org.junit.jupiter.api.*;

public abstract class BaseTest {

    @BeforeEach
    void setUp() {
        PlaywrightManager.init();
    }

    @AfterEach
    void tearDown(TestInfo info) {
        try {
            ScreenshotUtil.take(PlaywrightManager.page(), info.getDisplayName().replace(" ", "_"));
        } catch (Exception ignored) {}

        PlaywrightManager.close();
    }
}

File: src/main/kotlin/base/BaseTest.kt
package base

import driver.PlaywrightManager
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.TestInfo
import utils.ScreenshotUtil

abstract class BaseTest {

    @BeforeEach
    fun setUp() {
        PlaywrightManager.init()
    }

    @AfterEach
    fun tearDown(info: TestInfo) {
        try {
            ScreenshotUtil.take(PlaywrightManager.getPage(), info.displayName.replace(" ", "_"))
        } catch (_: Exception) {
        }

        PlaywrightManager.close()
    }
}
